Assets/LeapMotion/Core/Resources/HandData.cginc

```diff
@@ -0,0 +1,311 @@
+}
+
+float Leap_SqrDistToHand(float3 pos) {
+  // Fingertips
+	//float dist =     Leap_SqrDist(pos, mul(Leap_HandData_Preprocess_Matrix, _Leap_LH_Fingertips[0]));
+  //for (int i = 1; i < 5; i++) {
+	//  dist = min(dist, Leap_SqrDist(pos, mul(Leap_HandData_Preprocess_Matrix, _Leap_LH_Fingertips[i])));
+  //}
+  //for (int j = 0; j < 5; j++) {
+	//  dist = min(dist, Leap_SqrDist(pos, mul(Leap_HandData_Preprocess_Matrix, _Leap_RH_Fingertips[j])));
+  //}
+  float dist = 1000000000;
+
+  float handThickness = 0.00;
+
+  // Palm Triangles
+  for (int k = 0; k < 15; k += 3) {
+	  dist = min(dist, Leap_SqrDistTriangle_WithThickness(pos, mul(Leap_HandData_Preprocess_Matrix, _Leap_LH_PalmTriangles[k + 0]).xyz,
+                                                             mul(Leap_HandData_Preprocess_Matrix, _Leap_LH_PalmTriangles[k + 1]).xyz,
+                                                             mul(Leap_HandData_Preprocess_Matrix, _Leap_LH_PalmTriangles[k + 2]).xyz,
+                                                        (k < 3 ? handThickness : handThickness * 1.4)));
+  }
+  for (int l = 0; l < 15; l += 3) {
+	  dist = min(dist, Leap_SqrDistTriangle_WithThickness(pos, mul(Leap_HandData_Preprocess_Matrix, _Leap_RH_PalmTriangles[l + 0]).xyz,
+                                                             mul(Leap_HandData_Preprocess_Matrix, _Leap_RH_PalmTriangles[l + 1]).xyz,
+                                                             mul(Leap_HandData_Preprocess_Matrix, _Leap_RH_PalmTriangles[l + 2]).xyz,
+                                                        (l < 3 ? handThickness : handThickness * 1.4)));
+  }
+
+  // Finger Segments
+  for (int m = 0; m < 28; m += 2) {
+	  dist = min(dist, Leap_SqrDistLine_WithThickness(pos, mul(Leap_HandData_Preprocess_Matrix, _Leap_LH_FingerSegments[m + 0]).xyz,
+                                                         mul(Leap_HandData_Preprocess_Matrix, _Leap_LH_FingerSegments[m + 1]).xyz,
+                                                    handThickness));
+  }
+  for (int n = 0; n < 28; n += 2) {
+	  dist = min(dist, Leap_SqrDistLine_WithThickness(pos, mul(Leap_HandData_Preprocess_Matrix, _Leap_RH_FingerSegments[n + 0]).xyz,
+                                                         mul(Leap_HandData_Preprocess_Matrix, _Leap_RH_FingerSegments[n + 1]).xyz,
+                                                    handThickness));
+  }
+
+	return dist;
+}
+
+// ------
+// Planes
+// ------
+
+//float3 Leap_FingertipsDepthInPlane_AssumeHandsInLocalPlaneSpace() {
+//  float3 closest = _Leap_LH_Fingertips[0];
+//  for (int i = 1; i < 5; i++) {
+//    if (_leap_LH_Fingertips[i].z < )
+//  }
+//}
+
+// Copied-- plane displacement shader
+// TODO: Rename, potentially refactor
+
+// Uses dot product to get the component distance along the normal.
+// Also sets the parallel and perpendicular vectors.
+float distanceToPlane(float3 normal, float4 point1, float4 point2, out float3 perp, out float3 para) {
```

Comment:
Does this belong up with the other distance functions?  It has a different naming convention than the rest.

Reply:
Will rename and move